modules/mediawiki/files/bin/deploy-mediawiki

```diff
@@ -12,35 +12,55 @@ def run(args, start):
     synced = loginfo['servers']
     del loginfo['servers']
     text = f'starting deploy of "{str(loginfo)}" to {synced}'
+    rsyncpaths = []
+    rsyncfiles = []
     if not args.nolog:
         os.system(f'/usr/local/bin/logsalmsg {text}')
     else:
         print(text)
     if args.config:
         exitcodes.append(os.system('sudo -u www-data cp -r /srv/mediawiki-staging/config/* /srv/mediawiki/config/'))
+        rsyncpaths.append('/srv/mediawiki/config')
     if args.world:
         exitcodes.append(os.system('sudo -u www-data cp -r /srv/mediawiki-staging/w/* /srv/mediawiki/w/'))
+        rsyncpaths.append('/srv/mediawiki/w')
     if args.files:
         files = str(args.files).split(',')
         for file in files:
             exitcodes.append(os.system(f'sudo -u www-data cp /srv/mediawiki-staging/{file} /srv/mediawiki/{file}'))
+            rsyncfiles.append(file)
     if args.folders:
         folders = str(args.folders).split(',')
         for folder in folders:
             exitcodes.append(os.system(f'sudo -u www-data cp -r /srv/mediawiki-staging/{folder}/* /srv/mediawiki/{folder}/'))
+            rsyncpaths.append(f'/srv/mediawiki/{folder}')
     if args.l10n:
         exitcodes.append(os.system('sudo -u www-data php /srv/mediawiki/w/maintenance/mergeMessageFileList.php --wiki=loginwiki --output /srv/mediawiki/config/ExtensionMessageFiles.php'))
         exitcodes.append(os.system('sudo -u www-data php /srv/mediawiki/w/maintenance/rebuildLocalisationCache.php --wiki=loginwiki'))
     if args.gitinfo:
         exitcodes.append(os.system('sudo -u www-data php /srv/mediawiki/w/extensions/MirahezeMagic/maintenance/rebuildVersionCache.php --use-staging --wiki=loginwiki'))
     if args.servers == 'skip':
-        print('very good because I have not wrote the rsync client')
+        print('Sync skipped. Mediawiki deploy has not passed canary stage.')
+        sync = False
     elif args.servers == 'all':
-        print('This would deploy everywhere')
+        serverlist = ['mw8', 'mw9', 'mw10', 'mw12', mw13', 'mwtask1']
```

Comment:
mw13 needs a '

Reply:
Also does test3 need added here or not?

Reply:
No, it's handled seperately to production or we'd have a nightmare when testing upgrades.